Sources/Host/POSIX/FileIO.cpp

```diff
@@ -37,6 +38,29 @@ ErrorCode FileIO::CloseFile(int fd) {
   return kSuccess;
 }
 
+ErrorCode FileIO::FileRead(int fd, std::string &buf, uint64_t &count,
+                           uint64_t offset) {
+  if (offset > std::numeric_limits<off_t>::max()) {
+    return kErrorInvalidArgument;
+  }
+
+  off_t offArg = static_cast<off_t>(offset);
+  size_t countArg = static_cast<size_t>(count);
+
+  buf.resize(countArg);
+
+  ssize_t nRead = ::pread(fd, &buf[0], countArg, offArg);
```

Comment:
Yeah you're right, we aren't really consistent on this.

If we were using `U8Vector` though, we could do `buf.data()` everywhere,
without having to cast anything :)

On Wed, May 25, 2016 at 3:10 PM, kokarez <notifications@github.com> wrote:

> In Sources/Host/POSIX/FileIO.cpp
> <https://github.com/facebook/ds2/pull/353#discussion_r64662422>:
>
> > @@ -37,6 +38,29 @@ ErrorCode FileIO::CloseFile(int fd) {
> >    return kSuccess;
> >  }
> >
> > +ErrorCode FileIO::FileRead(int fd, std::string &buf, uint64_t &count,
> > +                           uint64_t offset) {
> > +  if (offset > std::numeric_limits<off_t>::max()) {
> > +    return kErrorInvalidArgument;
> > +  }
> > +
> > +  off_t offArg = static_cast<off_t>(offset);
> > +  size_t countArg = static_cast<size_t>(count);
> > +
> > +  buf.resize(countArg);
> > +
> > +  ssize_t nRead = ::pread(fd, &buf[0], countArg, offArg);
>
> Yea, in an other commit you use a cast to bypass that:
>
> const_cast<char *>(_stopInfo.debugString.c_str())
>
> I just point out this to have some *standart* usage. Not sure which one
> is better.
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://github.com/facebook/ds2/pull/353/files/00af9f8f51ac68b234e2901711792e11cf956058..56b2e8ce70a38350cc279f43e0a89254ee8209d9#r64662422>
>



-- 
Stephane Sezer